deepchecks/vision/datasets/classification/mnist.py

```diff
@@ -181,33 +195,134 @@ def load_model(pretrained: bool = True, path: pathlib.Path = None) -> 'MockMnist
 
     torch.save(model.state_dict(), path)
     model.eval()
-    return MockMnist(model)
+    return MockModel(model, dev)
 
 
-class MockMnist:
+class MockModel:
     """Class of MNIST model that returns cached predictions."""
 
-    def __init__(self, real_model):
-        self.real_model = real_model
+    def __init__(self, real_model, device):
+        self.device = device
+        self.real_model = real_model.to(device)
         with open(MNIST_DIR / 'static_predictions.pickle', 'rb') as handle:
             predictions = pickle.load(handle)
-        self.cache = {key: torch.tensor(value) for key, value in predictions.items()}
+        self.cache = {key: torch.tensor(value).to(device) for key, value in predictions.items()}
 
     def __call__(self, batch):
         results = []
         for img in batch:
             hash_key = hash_image(img)
             if hash_key not in self.cache:
-                prediction = self.real_model(torch.stack([img]))[0]
+                prediction = self.real_model(torch.stack([img]).to(self.device))[0]
                 prediction = nn.Softmax()(prediction).detach()
-                self.cache[hash_key] = prediction
+                self.cache[hash_key] = prediction.to(self.device)
             results.append(self.cache[hash_key])
         return torch.stack(results)
 
 
-class MNIST(datasets.MNIST):
+class IterableTorchMnistDataset(IterableDataset):
```

Comment:
Can't this also subclass Mnist but be iterable? Has to download a different copy of mnist by itself?

Reply:
it downloads uses same copy